spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/BlockingQueueConsumer.java

```diff
@@ -808,8 +811,20 @@ public void handleDelivery(String consumerTag, Envelope envelope, AMQP.BasicProp
 			try {
 				if (BlockingQueueConsumer.this.abortStarted > 0) {
 					if (!BlockingQueueConsumer.this.queue.offer(new Delivery(consumerTag, envelope, properties, body),
-							BlockingQueueConsumer.this.shutdownTimeout, TimeUnit.MILLISECONDS)) { // NOSONAR
-						// ignore - we're aborting anyway
+							BlockingQueueConsumer.this.shutdownTimeout, TimeUnit.MILLISECONDS)) {
+						try {
+							// Defensive - should never happen
+							BlockingQueueConsumer.this.queue.clear();
+							getChannel().basicNack(envelope.getDeliveryTag(), true, true);
+							getChannel().basicCancel(consumerTag);
+							RabbitUtils.setPhysicalCloseRequired(true);
+							getChannel().close();
+						}
+						catch (Exception e) {
+							if (logger.isDebugEnabled()) {
+								logger.debug("Error performing 'basicCancel'", e);
+							}
```

Comment:
This is an exception. Why just don't let to bubble as is?
More over we may have an exception not only on the `basicCancel` here.